File: Dictionary_LinkedList_Java/DictionaryTest.java
class DictionaryTest{
	public static void main(String[] args){
		Dictionary test = new Dictionary();

		//testing delete()
		//test.delete("4");
		//result: Throws exception 
		//test 1 ends here.
//----------------------------------------------------------------------------------------

		//testing isEmpty()
		System.out.println( "Dictionary empty: " + test.isEmpty() );

		//dictionary test contains nothing up to this point.
		//test 2 ends here.
//----------------------------------------------------------------------------------------

		//check spcial cases. adding and deleting in first node.
		test.insert("1", "z");
		test.delete("1");
		System.out.println("Dictionary contains:\n" + test);

		//testing insert() to an existing dictionary
		test.insert("10", "z");	
		test.insert("6", "t");
		test.insert("7", "s");
		System.out.println("Dictionary contains:\n" + test);

		//testing delete() to more than one element in dictionary
		test.delete("10");
		test.delete("6");
		System.out.println("Dictionary contains:\n" + test);

		//dictionary test contains key: 7 up to this point
		//test 3 ends here.
//----------------------------------------------------------------------------------------

		//retest isEmpty() & test lookup()
		System.out.println( "Dictionary empty: " + test.isEmpty() );

		test.insert("3", "c");
		test.insert("4", "e");

		System.out.println("Dictionary contains:\n" + test);

		test.delete("4");

		if(test.lookup("4") == null)
			System.out.println("test.lookup(4) failed.");

		if(test.lookup("3") != null )
			System.out.println("test.lookup(3) success.");

		//dictionary test contains key: 7, 3, 4 up to this point
		//test 4 ends here.
//----------------------------------------------------------------------------------------

		//test makeEmpty and size()
		test.makeEmpty();

		System.out.println("Emptying dictionary...");
		System.out.println( "Dictionary empty: " + test.isEmpty() );
		System.out.println( "Dictionary size: " + test.size() );

		test.insert("4", "a");
		test.insert("5", "b");
		test.insert("6", "c");

		System.out.println("Inserted " + test.size() + " to dictionary." );
		System.out.println("New size: " + test.size() + "\n" );

		//dictionary test contains key: 4, 5, 6 up to this point
		//test 5 ends here.
//----------------------------------------------------------------------------------------
		//testing Exceptions but do not throw it.
		try{
			test.insert("4", "a");
		}
		catch(DuplicateKeyException e){
			System.out.println(e + "\nContinuing program...\n" );
		}

		try{
			test.delete("123");
		}
		catch(KeyNotFoundException e){
			System.out.println(e + "\nContinuing program...\n" );
		}
		//dictionary contains key: 4, 5, 6 up to this point 
		//test 6 ends here.
//----------------------------------------------------------------------------------------
		System.out.println("All ADT operations tested. Working 100%");
	}
}

File: Dictionary Class in C/DictionaryTest.c
//Alston Huang
//1471706
//CMPS012m
//A test that tests all variation of the ADT operations.
//DictionaryTest.c

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){

   Dictionary A = newDictionary();

//-----------------------------------------------------------------------------------------------
   //test an empty dictionary. Should return 1
   printf( "Dictionary Empty: %d\n", isEmpty(A) );

   //Dictionary A contains nothing up to this point
   //test 1 ends here

//-----------------------------------------------------------------------------------------------

   //testing inserts
   insert( A, "one", "foo" );
   insert( A, "two", "bar" );
   insert( A, "three", "blah");
   printf( "Successfully inserted three keys to Dictionary A\n");
   //test size. should return 3
   printf( "Dictionary Size: %d\n", size(A) );

   //Dictionary A contains keys: one, two, three
   //test 2 ends here
//-----------------------------------------------------------------------------------------------

   //testing a non-empty dictionary. should return 0
   printf( "Dictionary Empty: %d\n", isEmpty(A) );

   //Dictionary A contains keys: one, two, three
   //test 3 ends here
//-----------------------------------------------------------------------------------------------

   printDictionary(stdout, A);

   //testing delete on Dictionary containing multiple elements
   delete( A, "one" );
   delete( A, "two" );

   //test delete on Dictionary containing one element
   delete( A, "three" );
   printf("Successfuly deleted everything in dictionary using delete()\n");

   //re-test size on supposedly empty dictionary. should return 0
   printf("Dictionary Size: %d\n", size(A));

   //should print nothing
   printDictionary(stdout, A);

   //Dictionary A contains nothing up to here
   //test 4 ends here
//-----------------------------------------------------------------------------------------------

   insert( A, "one", "foo" );
   //test for makeEmpty();
   makeEmpty(A);
   printf("Making Dictionary Empty...\n");
   //printDictionary should print nothing, size shold be 0, isEmpty = 1
   printDictionary(stdout, A);
   printf("Dictionary Size: %d\nDictionary Empty: %d\n", size(A), isEmpty(A) );

   //Dictionary A contains nothing up to here
   //test 5 ends here
//-----------------------------------------------------------------------------------------------

   insert( A, "one", "foo" );
   //test lookup on dictionary containing one element
   printf( "Lookup(A, one): %s\n", lookup(A, "one") );

   insert( A, "two", "boo" );
   //test lookup on dictionary containing more than one element 
   printf( "Lookup(A, one): %s\nLookup(A, two): %s\n", lookup(A, "one"), lookup(A, "two") );

   //Dictionary A contains keys: one, two
   //test 6 ends here
//-----------------------------------------------------------------------------------------------

   printf("\nAll ADT operations for Dictionary.c working 100%%\n");

   freeDictionary(&A);

   return(EXIT_SUCCESS);
}